course6/toyapp-vivike/displaylib/objects.h

```diff
@@ -14,44 +15,70 @@ struct Color
 	void redder(int amount);
 };
 
-struct Vector2D
-{
-	int x;
-	int y;
-
-	void move(int xmove, int ymove);
-	
-	Vector2D operator +(Vector2D b) { return { x + b.x, y + b.y}; }
-	Vector2D operator +(int b) { return { x + b, y + b}; }	
-
-	Vector2D operator *(int b) { return { x * b, y * b}; }
-
-	Vector2D operator -(Vector2D r) { return { x - r.x, y - r.y}; }
-	Vector2D operator -(int r) { return { x - r, y - r}; }
-
-	Vector2D operator /(int r) { return { x / r, y / r}; }
-};
 
-struct Vector3D
+template<typename T, int length>
+struct Vector
 {
-	int x;
-	int y;
-	int z;
-
-	Vector3D operator +(Vector3D b) { return { x + b.x, y + b.y, z + b.z}; }
-	Vector3D operator +(int b) { return { x + b, y + b, z + b}; }	
-
-	Vector3D operator *(int b) { return { x * b, y * b, z * b}; }
-
-	Vector3D operator -(Vector3D r) { return { x - r.x, y - r.y, z - r.z}; }
-	Vector3D operator -(int r) { return { x - r, y - r, z - r}; }
-
-	Vector3D operator /(int r) { return { x / r, y / r, z / r}; }
+	int vec[length];
+
+	Vector<T, length> operator +(Vector& b) 
```

Comment:
Inside of the class' definition, you don't need to provide template arguments for the class. ```Vector``` will mean ```Vector<T, length>```